### PR TITLE
Support ghc-8.8.1

### LINKED ISSUE #78: Support GHC-8.8.1
TODO:
- [ ] Update Travis config
- [ ] Add `-Wmissing-deriving-strategies` to `.cabal` file ghc options

### PR DESCRIPTION
**Pull request by brandonhamilton**
Resolves #78

### CODE REVIEW

File: test/Test/TypeRep/MapProperty.hs [hunk 1]
```diff
@@ -83,7 +83,7 @@ test_DeleteInvariant = prop "invariantCheck (delete k b) == True" $ do
 -- it doesn't compare values so it's not true equality. But this should be
 -- enough for tests.
 newtype FpMap f = FpMap (TypeRepMap f)
-  deriving (Show, Semigroup, Monoid)
+  deriving stock (Show, Semigroup, Monoid)
```
Comment (chshersh) [inline test/Test/TypeRep/MapProperty.hs]: I think `newtyp` strategy is more appropriate here
```suggestion
  deriving newtype (Show, Semigroup, Monoid)
```

File: stack.yaml [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
 extra-deps:
-  - aeson-1.4.4.0
+  - dependent-map-0.2.4.0
   - dependent-sum-0.5
-  - hedgehog-1.0
-  - primitive-0.7.0.0
-  - tasty-hedgehog-1.0.0.1
-  - time-compat-1.9.2.2
+
+allow-newer: true
```
Comment (chshersh) [inline stack.yaml]: Do you remember why this `allow-newer` is needed?

File: test/Test/TypeRep/MapProperty.hs (hunk 1 above)
Comment (brandonhamilton) [reply test/Test/TypeRep/MapProperty.hs]: Yes, agreed

File: stack.yaml (hunk 2 above)
Comment (brandonhamilton) [reply stack.yaml]: Actually, it's not needed. I had put in it when trying to resolve the correct versions of dependencies and forgot to remove it ;)

**Pull request merged**